subsys/caf/events/Kconfig.keep_alive_event

```diff
@@ -4,15 +4,15 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config PELION_CLIENT_KEEP_ALIVE_EVENTS
+config CAF_KEEP_ALIVE_EVENTS
```

Comment:
We have just introduced Pelion client in v1.6.0, and now we are renaming Kconfig symbols which will require all those Kconfig settings to be renamed for end-users if they have made there own configs.

Could we potentially just deprecate the old names for one more release, and have a direct translation, like:
```
config PELION_CLIENT_KEEP_ALIVE_EVENTS
	bool "Enable keep alive events (deprecated)"
	select CAF_KEEP_ALIVE_EVENTS
	help
	  Enable keep alive events.
	  This setting is deprecated, use CAF_KEEP_ALIVE_EVENTS instead.

```

Reply:
@Rallare and @OeySa what is your opinion on this ?

Reply:
I was trying to provide some solution that would generate warning if deprecated option is selected.
I really like how it is done in u-boot: https://github.com/openil/u-boot/blob/master/Kconfig, and then any deprecated option will depend on this setting. But this is HARD depreciation info, we wish to have a kind of SOFT information.

We can also generate a warning from code if we found that old option is selected.
I get your points, I will implement it in separate commit to allow just reverting to remove it.

But I believe we could implement NRF_DEPRECATED option in our repository and just depend on it all our options that we wish to remove and spend more time to provide good migration manual instead of pleasing the clients that are using NCS 1.6.0 to limit the number of changes they need to do migrating to NCS 1.7.0. It would make sense only if NCS 1.6.0 is LTS. Then we should keep all the changes until the next LTS.

The problem is I did not find any option to make a warning on Kconfig level. If we would just leave it the clients that are using NCS 1.6.0 would never see the warning and they would be surprised just in next release and we spend time and energy on something that finally gives no big gain to nobody. Good migration guide makes the migration in a form of option rename a matter of minutes or seconds.